swiftlang/swift-docc
# Make default availability inheritance behaviour customizable.

Bug/issue #, if applicable:  132980711

## Summary

Introduced a new Info.plist key, `CDDefaultAvailabilityOptions`, which enables customization of DocC's logic when applying default availability to symbols availability information. 

`CDDefaultAvailabilityOptions` is a dictionary that allows an inner key `InheritVersionNumber`. 

`InheritVersionNumber` is a boolean that allows the removal of the symbol default availability version on symbols that don't define an explicit availability annotation for the given platform.

## Dependencies

N/A

## Testing

Use the attached doc catalog.
[Availability.docc.zip](https://github.com/user-attachments/files/17051273/Availability.docc.zip)


Steps:
1. Preview the attached catalog and got to the symbol `Bar`
2. Check that the symbol `Bar` shows the platforms from the default availability but not the versions.
3. Check that `Foo` shows iOS with the version and the rest of the platforms without version.
4. Check that the framework page shows the default availability information.
5. Change the info plist `InheritVersionNumber ` to true and assert that the availability renders as expected.
6. Remove the key from the Info.plist and assert that the availability renders as expected.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
